rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/deduplication/TbMsgDeduplicationNode.java

```diff
@@ -154,7 +177,7 @@ private void processDeduplication(TbContext ctx, EntityId deduplicationId) {
                     }
                     if (resultMsg != null) {
                         deduplicationResults.add(TbMsg.newMsg(
-                                resultMsg.getQueueName(),
+                                queueName,
```

Comment:
For backward compatibility:

```suggestion
                                queueName != null ? queueName : resultMsg.getQueueName(),
```


Please add a test for this logic change.